pandas/tests/indexing/test_loc.py

```diff
@@ -711,3 +711,36 @@ def test_identity_slice_returns_new_object(self):
 
         original_series[:3] = [7, 8, 9]
         assert all(sliced_series[:3] == [7, 8, 9])
+
+    @pytest.mark.parametrize('indexer_type_1',
+                             (list, tuple, set, slice, np.ndarray, Series))
+    @pytest.mark.parametrize('indexer_type_2',
+                             (list, tuple, set, slice, np.ndarray, Series))
+    def test_loc_getitem_nested_indexer(self, indexer_type_1, indexer_type_2):
+
+        def convert_nested_indexer(indexer_type, keys):
+            if indexer_type == np.ndarray:
+                return np.array(keys)
+            if indexer_type == slice:
+                return slice(*keys)
+            return indexer_type(keys)
+
+        a = [10, 20, 30]
+        b = [1, 2, 3]
+        index = pd.MultiIndex.from_product([a, b])
+        df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])
+
+        keys = ([10, 20], [2, 3])
+        types = (indexer_type_1, indexer_type_2)
```

Comment:
I *think* you might be able to use ``np.ix_`` for this, we also have some utilities in pandas/core/indexing.py